scripts/dss_v2_apdapter.py

```diff
@@ -0,0 +1,489 @@
+        """
+        prefixes = self.get_prefix_list(self.args.bundle_uuid_prefix, self.args.bundle_uuid_start_prefix)
+
+        params = {
+            'replica': self.dss_src_replica,
+            'per_page': 500,  # limits how many bundles per request min=10 max=500
+        }
+        if prefixes:
+            params['prefix'] = prefixes.pop(0)
+        url_base = self.dss_endpoint + '/bundles/all'
+        url = url_base + '?' + parse.urlencode(params)
+
+        # Loop to request all pages of results for each prefix
+        while True:
+            log.info('Requesting list of bundles: %s', url)
+            response = requests.get(url)
+            response.raise_for_status()
+            response_json = response.json()
+            with ThreadPoolExecutor(max_workers=self.num_workers) as tpe:
+                future_to_bundle = {
+                    tpe.submit(self._process_bundle,
+                               bundle['uuid'],
+                               bundle['version']): f"{bundle['uuid']}.{bundle['version']}"
+                    for bundle in response_json['bundles']
+                }
+                for future in as_completed(future_to_bundle):
+                    bundle_fqid = future_to_bundle[future]
+                    try:
+                        future.result()
+                    except Exception as e:
+                        self.errors.append(f'Bundle {bundle_fqid} had Exception: {e}')
+                        f: Future
+                        for f in future_to_bundle:
+                            f.cancel()
+            if response_json.get('has_more'):
+                url = response_json.get('link')
+            elif prefixes:
+                params['prefix'] = prefixes.pop(0)
+                url = url_base + '?' + parse.urlencode(params)
+            else:
+                break
+
+    def _process_bundle(self, bundle_uuid: str, bundle_version: str):
+        """
+        Transfer all files from the given bundle to a GCS staging bucket.
+        """
+        if self.errors:
+            return
+        benchmarks = []
+        bundle_fqid = f'{bundle_uuid}_{bundle_version}'
+        log.info('Requesting Bundle: %s', bundle_fqid)
+        t1 = time.perf_counter()
+        bundle = self.mini_dss.get_bundle(uuid=bundle_uuid, version=bundle_version, replica=self.dss_src_replica)
+        benchmarks.append((time.perf_counter() - t1, f'to request bundle {bundle_fqid}'))
+
+        # First loop over the manifest entries to gather needed info to modify
+        # links.json and metadata files. Also use this loop to validate the
+        # files before starting the upload of this bundle.
+        schema_types = {}  # Mapping of file uuid to schema type
+        manifest_entries: MutableJSON = {}  # Mapping of file names to manifest entry
+        file_name_re = re.compile(r'^(\w+)_\d+\.json$')  # ex. 'cell_suspension_0.json'
+        t1 = time.perf_counter()
```

Comment:
Low priority: Just be sure you understand what perf_counter() is really for. time.time() would be perfectly adequate for this.